experiments/AMIP/moist_mpi_earth_dynamical_sea_ice/slab_ice/slab_init.jl

```diff
@@ -0,0 +1,213 @@
+    if h_ice > 0 # ice-covered 
+        F_base = p.F0_base * (T_ml - p.T_base)
+        ΔT_ml = -(F_base + ocean_qflux) * Δt / (p.h_ml * p.ρc_ml)
+        Δh_ice = (F_atm - F_base) * Δt / p.L_ice
+    else # ice-free
+        ΔT_ml = -(F_atm + ocean_qflux) * Δt / (p.h_ml * p.ρc_ml)
+        Δh_ice = 0
+    end
+
+    # adjust if transition to ice-covered 
+    if (T_ml[1] + ΔT_ml[1] < p.T_freeze)
+        Δh_ice = Δh_ice - (T_ml + ΔT_ml - p.T_freeze) * (p.h_ml * p.ρc_ml) / p.L_ice
+        ΔT_ml = p.T_freeze - T_ml
+    end
+
+    # adjust if transition to ice-free
+    if ((h_ice[1] > 0) & (h_ice[1] + Δh_ice[1] <= 0))
+        ΔT_ml = ΔT_ml - (h_ice + Δh_ice) * p.L_ice / (p.h_ml * p.ρc_ml)
+        Δh_ice = -h_ice
+    end
+
+    # solve for T_sfc
+    if (h_ice[1] + Δh_ice[1] > 0) #  surface is ice-covered
+        # if ice covered, solve implicity (for now one Newton iteration: ΔT_s = - F(T_s) / dF(T_s)/dT_s )
+        F_conductive = p.k_ice / (h_ice + Δh_ice) * (p.T_base - T_sfc)
+        ΔT_sfc = (-F_atm + F_conductive) / (p.k_ice / (h_ice + Δh_ice) + ∂F_atm∂T_sfc)
+        if (T_sfc[1] + ΔT_sfc[1] > p.T_freeze)
+            ΔT_sfc = p.T_freeze - T_sfc
+        end
+        # surface is ice-covered, so update T_sfc as ice surface temperature
+
+        T_sfc += FT(ΔT_sfc)
+    else # ice-free, so update T_sfc as mixed layer temperature
+        T_sfc = FT(T_ml + ΔT_ml)
+    end
+
+    T_ml += FT(ΔT_ml)
+    h_ice += FT(Δh_ice)
+
+    return (; T_ml = FT(T_ml), h_ice = FT(h_ice), F_base = FT(F_base), T_sfc = FT(T_sfc))
+end
+
+get_∂F_atm∂T_sfc(p, T_sfc, Ya) = @. FT(4) * (FT(1) - p.α) * p.σ * T_sfc^3 + Ya.∂F_aero∂T_sfc
+
+function ∑tendencies_ice_stub(du, u, p, t)
+    dY = du
+    Y = u
+    FT = eltype(dY)
+
+    if p.Ya.prescribed_sic_data !== nothing # Prognostic eqn for T_sfc
+
+        @show "pres SIC"
+        params = p.Ya.params
+        F_aero = p.Ya.F_aero
+        F_rad = p.Ya.F_rad
+        ice_mask = p.Ya.ice_mask
+
+        F_conductive = @. params.k_ice / (params.h) * (params.T_base - Y.T_sfc) .* FT(0)
+        rhs = @. (-F_aero - F_rad + F_conductive) / (params.h * params.ρ * params.c)
+
+        # ∂F_atm∂T_sfc = get_∂F_atm∂T_sfc(p, T_sfc, Ya)
+        # rhs = @. (-F_aero - F_rad + F_conductive) / (p.k_ice / params.h + ∂F_atm∂T_sfc)
+        parent(dY.T_sfc) .= apply_mask.(parent(ice_mask), >, parent(rhs), FT(0), FT(0))
+    else
+        solve_ice!((; u = u, p = p), p.Ya.Δt) # timestepping outside of DeffEq (but DeffEq still used here for saving vars in `integ.sol`)
+
+        @. dY.T_ml = FT(0)
+        @. dY.h_ice = FT(0)
+        @. dY.T_sfc = FT(0)
+    end
+end
+
+function slab_ice_init(
+    ::Type{FT},
+    tspan;
+    stepper = Euler(),
+    nelements = 6,
+    npolynomial = 4,
+    dt = 0.02,
+    saveat = 1.0e10,
+    space = nothing,
+    mask = nothing,
+    prescribed_sic_data = nothing,
+    ocean_params = (; ρ = FT(1e3), c = FT(4e3), h = FT(1)),
```

Comment:
should these be enforced to be consistent with the slab_ocean params being used?

Reply:
True, I made it a non-keyword argument, so there is no accidental imposition of the default ocean parameters.